test/framework/src/main/java/org/elasticsearch/cluster/coordination/CoordinationStateTestCluster.java

```diff
@@ -102,6 +105,26 @@ public static long value(ClusterState clusterState) {
         }
 
         void reboot() {
+            if (localNode.isMasterNode() == false && rarely()) {
```

Comment:
should we do something similar for CoordinatorTests to check liveness aspects of this?

Reply:
Sure, I added a thing to arbitrarily roll back the state on data nodes in 0fa253d2a82cdf00324861e241b183098a321418.